File: app/src/main/java/eu/span/dev/rxjava/warcraft/model/Boss.java
package eu.span.dev.rxjava.warcraft.model;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Boss {

    @SerializedName("id")
    @Expose
    public Integer id;
    @SerializedName("name")
    @Expose
    public String name;
    @SerializedName("urlSlug")
    @Expose
    public String urlSlug;
    @SerializedName("description")
    @Expose
    public String description;
    @SerializedName("zoneId")
    @Expose
    public Integer zoneId;
    @SerializedName("availableInNormalMode")
    @Expose
    public Boolean availableInNormalMode;
    @SerializedName("availableInHeroicMode")
    @Expose
    public Boolean availableInHeroicMode;
    @SerializedName("health")
    @Expose
    public Integer health;
    @SerializedName("heroicHealth")
    @Expose
    public Integer heroicHealth;
    @SerializedName("level")
    @Expose
    public Integer level;
    @SerializedName("heroicLevel")
    @Expose
    public Integer heroicLevel;
    @SerializedName("journalId")
    @Expose
    public Integer journalId;
    @SerializedName("npcs")
    @Expose
    public List<Npc> npcs = null;

}

File: app/src/main/kotlin/warcraft/model/boss/Boss.kt
package warcraft.model.boss

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Boss {

    @SerializedName("id")
    @Expose
    var id: Int? = null;

    @SerializedName("name")
    @Expose
    lateinit var name: String

    @SerializedName("urlSlug")
    @Expose
    lateinit var urlSlug: String

    @SerializedName("description")
    @Expose
    lateinit var description: String

    @SerializedName("zoneId")
    @Expose
    var zoneId: Int? = null

    @SerializedName("availableInNormalMode")
    @Expose
    var availableInNormalMode: Boolean? = null

    @SerializedName("availableInHeroicMode")
    @Expose
    var availableInHeroicMode: Boolean? = null

    @SerializedName("health")
    @Expose
    var health: Int? = null

    @SerializedName("heroicHealth")
    @Expose
    var heroicHealth: Int? = null

    @SerializedName("level")
    @Expose
    var level: Int? = null

    @SerializedName("heroicLevel")
    @Expose
    var heroicLevel: Int? = null
    @SerializedName("journalId")
    @Expose
    var journalId: Int? = null

    @SerializedName("npcs")
    @Expose
    lateinit var npcs: List<Npc>
}
